File: Codewars/Python/Kyu5_Rot13.py
import string

def rot13(message):
    alpha = string.ascii_uppercase
    result = ""
    for char in message:
        if char in alpha.lower():
            result += alpha[(alpha.lower().index(char) +13) % 26].lower()
        elif char in alpha:
            result += alpha[(alpha.index(char) +13) % 26]
        else:
            result += char
    return result
    

File: Codewars/Go/Kyu5_ROT13.go
package kata

import (
  "unicode"
)

func Rot13(msg string) string {
  var abj string = "abcdefghijklmnopqrstuvwxyz"
	var dummyPosition int
	var result string

	for _, charMsg := range msg {
		// iterate every message character
		if unicode.IsLetter(charMsg) {
			for index, charAbj := range abj {
				// iterate to find abjad index of letter
				if unicode.ToLower(charMsg) == charAbj {
					dummyPosition = index

					dummyPosition += 13 // shift position
					dummyPosition %= 26 // implement limit

					// assign new letter
					if unicode.IsUpper(charMsg) {
						result += string(unicode.ToUpper(rune(abj[dummyPosition])))
						continue
					}

					result += string(abj[dummyPosition])
					break
				}
			}

			continue
		}

		result += string(charMsg)
	}

	return result
}